public/app/features/explore/RichHistory/RichHistoryContent.tsx

```diff
@@ -0,0 +1,244 @@
+  { label: 'Datasource Z-A', value: SortOrder.DatasourceZA },
+];
+
+interface RichHistoryContentProps {
+  queries: RichHistoryQuery[];
+  sortOrder: SortOrder;
+  activeDatasourceOnly: boolean;
+  activeDatasourceInstance: string;
+  datasourceFilters: SelectableValue[] | null;
+  onlyStarred: boolean;
+  retentionPeriod: number;
+  onChangeSortOrder: (sortOrder: SortOrder) => void;
+  onChangeRichHistoryProperty: (ts: number, property: string, updatedProperty?: string) => void;
+  onSelectDatasourceFilters: (value: SelectableValue[] | null) => void;
+}
+
+const getStyles = stylesFactory((theme: GrafanaTheme, onlyStarred: boolean) => {
+  const bgColor = theme.isLight ? theme.colors.gray5 : theme.colors.dark4;
+
+  /* 134px is based on the width of the Query history tabs bar, so the content is aligned to right side of the tab */
+  const cardWidth = onlyStarred ? '100%' : '100% - 134px';
+  return {
+    container: css`
+      display: flex;
+      .label-slider {
+        font-size: ${theme.typography.size.sm};
+        &:last-of-type {
+          margin-top: ${theme.spacing.lg};
+        }
+        &:first-of-type {
+          margin-top: ${theme.spacing.sm};
+          font-weight: ${theme.typography.weight.semibold};
+          margin-bottom: ${theme.spacing.xs};
+        }
+      }
+    `,
+    containerContent: css`
+      width: calc(${cardWidth});
+    `,
+    containerSlider: css`
+      width: 125px;
+      margin-right: ${theme.spacing.sm};
+      .slider {
+        bottom: 10px;
+        height: 200px;
+        width: 125px;
+        padding: ${theme.spacing.xs} 0;
+      }
+    `,
+    slider: css`
+      height: 300px;
+      position: absolute;
+    `,
+    selectors: css`
+      display: flex;
+      justify-content: space-between;
+    `,
+    multiselect: css`
+      width: 60%;
+      .gf-form-select-box__multi-value {
+        background-color: ${bgColor};
+        padding: ${theme.spacing.xxs} ${theme.spacing.xs} ${theme.spacing.xxs} ${theme.spacing.sm};
+        border-radius: ${theme.border.radius.sm};
+      }
+    `,
+    sort: css`
+      width: 170px;
+    `,
+    sessionName: css`
+      display: flex;
+      align-items: flex-start;
+      justify-content: flex-start;
+      margin-top: ${theme.spacing.lg};
+      h4 {
+        margin: 0 10px 0 0;
+      }
+    `,
+    heading: css`
+      font-size: ${theme.typography.heading.h4};
+      margin: ${theme.spacing.md} ${theme.spacing.xxs} ${theme.spacing.sm} ${theme.spacing.xxs};
+    `,
+  };
+});
+
+export function RichHistoryContent(props: RichHistoryContentProps) {
+  const {
+    datasourceFilters,
+    onSelectDatasourceFilters,
+    queries,
+    onlyStarred,
+    onChangeSortOrder,
+    sortOrder,
+    onChangeRichHistoryProperty,
+    activeDatasourceOnly,
+    activeDatasourceInstance,
+    retentionPeriod,
+  } = props;
+
+  const [sliderRetentionFilter, setSliderRetentionFilter] = useState<number[]>([0, retentionPeriod]);
```

Comment:
I think you can use `[number, number]` as a type to specify it will always be just tuple of 2 numbers.

Reply:
👍